src/impls.rs

```diff
@@ -279,6 +280,17 @@ where
     }
 }
 
+impl<T> TypeInfo for Arc<T>
+where
+    T: TypeInfo + ?Sized + 'static,
+{
+    type Identity = T;
+
+    fn type_info() -> Type {
+        Self::Identity::type_info()
```

Comment:
We may want to revisit the idea of adding some extra information about transparent wrapper types like this, e.g. for `Box` to avoid https://github.com/paritytech/subxt/blob/c08eb6c8f1ae1ea61964196733008cc3a4bfb627/codegen/src/types/composite_def.rs#L307